src/watertap_contrib/seto/unit_models/surrogate/lt_med_surrogate.py

```diff
@@ -356,354 +376,388 @@ def temp_last_effect(b):
-                    * b.temp_last_effect
-                    * b.recovery_vol_phase[0, "Liq"]
-                    * specific_area_coeffs[b.number_effects.value][59]
+                    * self.temp_last_effect
+                    * self.recovery_vol_phase[0, "Liq"]
+                    * specific_area_coeffs[num_effect][59]
                     + temp_steam**2
-                    * b.temp_last_effect**2
-                    * specific_area_coeffs[b.number_effects.value][60]
-                    + temp_steam**3 * specific_area_coeffs[b.number_effects.value][61]
+                    * self.temp_last_effect**2
+                    * specific_area_coeffs[num_effect][60]
+                    + temp_steam**3 * specific_area_coeffs[num_effect][61]
                     + temp_steam**3
                     * feed_conc_ppm
-                    * specific_area_coeffs[b.number_effects.value][62]
+                    * specific_area_coeffs[num_effect][62]
                     + temp_steam**3
-                    * b.recovery_vol_phase[0, "Liq"]
-                    * specific_area_coeffs[b.number_effects.value][63]
+                    * self.recovery_vol_phase[0, "Liq"]
+                    * specific_area_coeffs[num_effect][63]
                     + temp_steam**3
-                    * b.temp_last_effect
-                    * specific_area_coeffs[b.number_effects.value][64]
-                    + 1 * specific_area_coeffs[b.number_effects.value][65]
-                    + temp_steam**4 * specific_area_coeffs[b.number_effects.value][66]
-                    + b.temp_last_effect**4
-                    * specific_area_coeffs[b.number_effects.value][67]
-                    + b.recovery_vol_phase[0, "Liq"] ** 4
-                    * specific_area_coeffs[b.number_effects.value][68]
-                    + feed_conc_ppm**4
-                    * specific_area_coeffs[b.number_effects.value][69]
+                    * self.temp_last_effect
+                    * specific_area_coeffs[num_effect][64]
+                    + 1 * specific_area_coeffs[num_effect][65]
+                    + temp_steam**4 * specific_area_coeffs[num_effect][66]
+                    + self.temp_last_effect**4 * specific_area_coeffs[num_effect][67]
+                    + self.recovery_vol_phase[0, "Liq"] ** 4
+                    * specific_area_coeffs[num_effect][68]
+                    + feed_conc_ppm**4 * specific_area_coeffs[num_effect][69]
                 )
 
+        # Surrogate equations were built for 3,6,9,12,14 effects
+        # For intermediate number of effects (4,5,7,8,10,11,13), linear interpolation is adopted
+        @self.Constraint(doc="Gain output ratio surrogate equation")
+        def eq_gain_output_ratio(b):
+            if b.config.number_effects in [3, 6, 9, 12, 14]:
+                return b.gain_output_ratio == _get_gain_output_ratio(
+                    b.config.number_effects
+                )
+            else:  # number_effects in [4,5,7,8,10,11,13]
```

Comment:
Should have another conditional that prevents <3 effects and >14 effects. We can raise an exception to be more stringent about the surrogate model's valid range.

Reply:
Sure, the exceptions are added.